### ISSUE TITLE
Accessibility: Incorrect usage of `<fieldset>`

### ISSUE THREAD

**Issue by GuySartorelli** [silverstripe/silverstripe-admin#1959]
[WCAG 2.2 A F43](https://www.w3.org/TR/WCAG20-TECHS/F43.html) 1.3.1
[WCAG 2.2 A H71](https://www.w3.org/TR/WCAG-TECHS/H71.html) 3.3.2

A `<fieldset>` element should be used to group logically-related field controls - for example a group of checkboxes that will submit to the same-named field, or a group of related fields like for an address (street name, city, country, etc).

Currently we use `<fieldset>` to just contain all of the fields for a form.

Also, while `<legend>` is optional in the HTML spec, for accessibility purposes "the first element inside the `fieldset` must be a `legend element`, which provides a label or description for the group".

Currently legends are optional and usually not set.

Lastly, `<fieldset>` should not exist if there are no _visible/interactable_ field controls. i.e. a fieldset that has only hidden fields is not valid.

## Notes

- Removing the fieldset and requiring a legend are both probably breaking changes. They definitely would be in the frontend, but probably are in the CMS as well. Caution is advised. This may be better to leave for a major release.
- Whether we change this in the CMS in a minor release or not, we should still endeavour to change it also in the frontend templates for a major release. Otherwise we're promoting bad accessibility practices in the frontend.
- Other than as the element that wraps _all_ fields in the form, `fieldset` is also used in `SelectionGroup` (for groups of checkboxes and radio buttons), and optionally in `CompositeField` (default div, optionally fieldset), along with a handful of other misc usages.
- The main form templates are:
  - `templates/SilverStripe/Admin/Includes/LeftAndMain_EditForm.ss` in admin
  - `themes/cms-forms/templates/SilverStripe/Forms/Includes/Form.ss` in admin
  - `templates/SilverStripe/CMS/Controllers/Includes/CMSMain_AddForm.ss` in cms
  - `templates/SilverStripe/CMS/Controllers/Includes/CMSMain_EditForm.ss` in cms
  - `templates/SilverStripe/CMS/Search/Includes/SearchForm.ss` in cms
  - `templates/SilverStripe/Forms/Includes/Form.ss` in framework (frontend)
  - `templates/SilverStripe/Security/MemberAuthenticator/Includes/CMSMemberLoginForm.ss` in framework (which along with its functionality probably should move to `silverstripe/admin`)
  - `templates/SilverStripe/SiteConfig/Includes/SiteConfigLeftAndMain_EditForm.ss` in siteconfig
  - `templates/SilverStripe/UserForms/Form/UserForm.ss` in userforms (frontend)
- Check both entwine and react forms

**Comment by SimulatedPanda** [comment]
As this contains breaking changes, please mark it for inclusion in the next CMS major release rather than completing this as part of a Minor release. Thank you
